src/Symfony/Component/Notifier/Bridge/Mattermost/MattermostTransportFactory.php

```diff
@@ -26,16 +27,22 @@ final class MattermostTransportFactory extends AbstractTransportFactory
     public function create(Dsn $dsn): TransportInterface
     {
         $scheme = $dsn->getScheme();
+
+        if ('mattermost' !== $scheme) {
+            throw new UnsupportedSchemeException($dsn, 'mattermost', $this->getSupportedSchemes());
+        }
+
         $token = $this->getUser($dsn);
         $channel = $dsn->getOption('channel');
-        $host = $dsn->getHost();
-        $port = $dsn->getPort();
 
-        if ('mattermost' === $scheme) {
-            return (new MattermostTransport($token, $channel, $this->client, $this->dispatcher))->setHost($host)->setPort($port);
+        if (!$channel) {
+            throw new IncompleteDsnException('Missing channel.', $dsn->getOriginalDsn());
```

Comment:
Could you please add a test for this case?

Reply:
Sure

Reply:
Done